pennylane/qnode.py

```diff
@@ -819,11 +819,7 @@ def __call__(self, *args, **kwargs):  # pylint: disable=too-many-branches, too-m
 
         if old_interface == "auto":
             self.interface = qml.math.get_interface(*args, *list(kwargs.values()))
-            if self.device is not self._original_device:
-                warnings.warn(
-                    "The device was switched during the call of the QNode, to avoid this behaviour define "
-                    "an interface argument instead of auto."
-                )
+            self.device.tracker = self._original_device.tracker

```

Comment:
maybe we can add a basic check for if the user added anything to the device? I've already confirmed that this is empty in the normal case:
```pycon
>>> dev = qml.device('default.qubit', wires=1)
>>> [set(vars(dev)) - set(vars(qml.device('default.qubit.'+suffix, wires=1))) for suffix in ["torch", "tf", "autograd", "jax"]]
[set(), set(), set(), set()]

```
```suggestion
            self.device.tracker = self._original_device.tracker
            if set(vars(self._original_device)) - set(vars(self.device)):
                warnings.warn(
                    "The device was switched during the call of the QNode, to avoid this behaviour define "
                    "an interface argument instead of auto."
                )
```

Reply:
we can ignore this, just an idea but tbh I don't think it'll add much value